## DATA EXPLORATION

This section shows the inital data exploration and cleaning.

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('../data/raw/historical_stock.csv', index_col=0, header=[0,1])

data.columns = [f"{col[0]}_{col[1]}" for col in data.columns]

df = pd.DataFrame(data)

df.head(10)

,ITX.MC_Close,AIR.PA_Close,SAN_Close,IBE.MC_Close,SIE.DE_Close
Date,,,,,
2000-01-03,NaN,NaN,2.925987,1.043016,31.938948
2000-01-04,NaN,NaN,2.863063,1.026695,31.058708
2000-01-05,NaN,NaN,2.831601,1.025141,29.803812
2000-01-06,NaN,NaN,2.831601,1.025141,28.443289
2000-01-07,NaN,NaN,2.941719,1.033690,29.587551
2000-01-10,NaN,NaN,2.863063,1.025918,31.347935
2000-01-11,NaN,NaN,2.800138,1.019700,31.564211
2000-01-12,NaN,NaN,2.737215,1.014260,31.229753
2000-01-13,NaN,NaN,2.752946,1.005710,32.441948


In [3]:
df.info() 
df.isnull().sum()

<class 'pandas.DataFrame'>
Index: 6916 entries, 2000-01-03 to 2026-09-17
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   ITX.MC_Close  6494 non-null   float64
 1   AIR.PA_Close  6427 non-null   float64
 2   SAN_Close     6717 non-null   float64
 3   IBE.MC_Close  6857 non-null   float64
 4   SIE.DE_Close  6825 non-null   float64
dtypes: float64(5)
memory usage: 324.2+ KB


ITX.MC_Close    422
AIR.PA_Close    489
SAN_Close       199
IBE.MC_Close     59
SIE.DE_Close     91
dtype: int64

The *historical_stock.csv* file has 6916 entries, from **2000-01-01** to **2026-09-17**. It is easyly observable that at least 59 days are null in every column, those days need to be assessed, as well as some other faulty registries clearly shown in the information above. 

In [4]:
null_dates = df.isnull()

null_dates = null_dates[null_dates.any(axis=1)]
null_dates = null_dates.index.tolist()
print(null_dates)
print(f"Number of null dates: {len(null_dates)}")
print(f"Percentage of null dates: {len(null_dates)/len(df)*100:.2f}%")

['2000-01-03', '2000-01-04', '2000-01-05', '2000-01-06', '2000-01-07', '2000-01-10', '2000-01-11', '2000-01-12', '2000-01-13', '2000-01-14', '2000-01-17', '2000-01-18', '2000-01-19', '2000-01-20', '2000-01-21', '2000-01-24', '2000-01-25', '2000-01-26', '2000-01-27', '2000-01-28', '2000-01-31', '2000-02-01', '2000-02-02', '2000-02-03', '2000-02-04', '2000-02-07', '2000-02-08', '2000-02-09', '2000-02-10', '2000-02-11', '2000-02-14', '2000-02-15', '2000-02-16', '2000-02-17', '2000-02-18', '2000-02-21', '2000-02-22', '2000-02-23', '2000-02-24', '2000-02-25', '2000-02-28', '2000-02-29', '2000-03-01', '2000-03-02', '2000-03-03', '2000-03-06', '2000-03-07', '2000-03-08', '2000-03-09', '2000-03-10', '2000-03-13', '2000-03-14', '2000-03-15', '2000-03-16', '2000-03-17', '2000-03-20', '2000-03-21', '2000-03-22', '2000-03-23', '2000-03-24', '2000-03-27', '2000-03-28', '2000-03-29', '2000-03-30', '2000-03-31', '2000-04-03', '2000-04-04', '2000-04-05', '2000-04-06', '2000-04-07', '2000-04-10', '2000

Being the percentage of null dates as low as $4\%$, these registries can be deleted without compromising the integrity of this study. 

In [5]:
df.dropna(inplace=True)

df.to_csv('../data/processed/historical_stock_cleaned.csv')